## AdaBoost Regression

In [117]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.ensemble import AdaBoostRegressor
from sklearn.model_selection import train_test_split
from sklearn.cross_validation import cross_val_score
import pickle

In [108]:
train_data = pd.read_csv("training/goalie.csv")

In [109]:
rng = 42
X = train_data
y = pd.read_csv("ground_truth/goalie_y.csv") 

In [110]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.15, random_state = 10) 

In [111]:
regr_2 = AdaBoostRegressor(DecisionTreeRegressor(max_depth=4),
                          n_estimators=300, random_state=rng)

In [ ]:
model2 = regr_2.fit(X, y)

In [ ]:
pickle.dump(regr_2,open("ada_boost_regressor.p","wb"))

In [118]:
scores2 = cross_val_score(model2,X_train,y_train,cv=5)
print( scores2)

/home/aayush/.local/lib/python3.5/site-packages/sklearn/utils/validation.py:578: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/home/aayush/.local/lib/python3.5/site-packages/sklearn/utils/validation.py:578: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/home/aayush/.local/lib/python3.5/site-packages/sklearn/utils/validation.py:578: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/home/aayush/.local/lib/python3.5/site-packages/sklearn/utils/validation.py:578: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of 

[0.89650457 0.9214935  0.92741885 0.78945223 0.90128941] [0.9540692  0.94998987 0.95367832 0.87748123 0.9614981 ]


In [113]:
y_2 = regr_2.predict(X)

In [115]:
print('Variance score: {}'.format(regr_2.score(X_test, y_test))) 

Variance score: 0.9739913063502766


## Results

For AdaBoost Regressor, accuracies of ratings of players are :

- **Goalkeeper** : 0.9739913063502766, **Cross-val** = [0.9540692  0.94998987 0.95367832 0.87748123 0.9614981]

- **Forward** : 0.9523572455211584, **Cross-val** = [0.87748769 0.8648086  0.91782871 0.92043024 0.85759141]

- **Defender** :0.9466706034891852,  **Cross-val** = [0.87581349 0.8426702  0.88626688 0.89040296 0.89876978]

- **Mid-fielder** :0.9364125788571634, **Cross-val** = [0.90837894 0.91648379 0.88919276 0.91691237 0.89910889]

- **Overall** :0.890911355435651, **Cross-val** = [0.85611865 0.87776786 0.86743015 0.87553897 0.84711675]